plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1814,7 +1814,7 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession sess
         }
 
         Map<String, Type> columnTypes = handle.getInputColumns().stream()
-                .collect(toImmutableMap(HiveColumnHandle::getName, column -> column.getHiveType().getType(typeManager)));
+                .collect(toImmutableMap(HiveColumnHandle::getName, column -> typeManager.getType(column.getHiveType().getTypeSignature())));
```

Comment:
Looks like `HiveType.getTypeSignature()` is deprecated for the same reason of not handling timestamp precision, so this change doesn't seem to improve things